File: client/src/types/JsonModels/Workouts/PreviousExerciseSetsModel.ts
import type { JsonModels } from "../../backend";

export interface PreviousExerciseSetsModel
{
	exerciseId: number;
	exerciseName: string;
	workoutId: number;
	workoutTitle: string;
	workoutStartedAt: string;
	sets: JsonModels.Workouts.PreviousExerciseSetModel[];
}

File: server/FitMate.Core/JsonModels/Workouts/PreviousExerciseSetsModel.cs
namespace FitMate.Core.JsonModels.Workouts;

public class PreviousExerciseSetsModel
{
    public long ExerciseId { get; set; }
    public string ExerciseName { get; set; } = string.Empty;
    public long WorkoutId { get; set; }
    public string WorkoutTitle { get; set; } = string.Empty;
    public DateTime WorkoutStartedAt { get; set; }
    public List<PreviousExerciseSetModel> Sets { get; set; } = [];
}
